### DISCUSSION TITLE
A little lost

### DISCUSSION THREAD

**Discussion by yuh8n** [seanap/Plex-Audiobook-Guide#74] [General]
I hate to send a question like this because of the incredible amount of time you have put into getting this together, but I'm lost and I'm not sure where to go forward with this configuration to get it working.  I'm struggling with understanding the workflow soup --> nuts - When I download my audiobook from cd and it has multiple MP3s do I create a directory somewhere that is named in a specific format and do I put that folder in temp or mergemp3?  

I just found out that the link in https://github.com/seanap/Plex-Audiobook-Guide for the auto merge takes you https://github.com/seanap/Auto-M4B-Tool#m4b-tool-automation which loads below your line that says "...outdated..." so I never even saw that until I was done setting up the docker following the steps on that page.  The new auto-M4B setup...I don't even really see any steps to set that up...does it use the docker that I already setup from the previous page?  I feel like I'm missing pieces to the puzzle.  As it sits right now, I can copy in mp3 files to mergemp3 and it will convert them, but a completed m4b file never ends up anywhere, it looks like it just ends up getting deleted.  

Sorry, I'm all over the place, but mp3tag, dropit, docker, bookcopy.sh it's all setup and now I need to figure out how to get the correct config for the auto convert/combine from linux/docker working correctly.  HELP.

**Comment by seanap** [comment]
Hey, no worries, audiobooks + plex has always been a bit of a workaround and can be quite confusing.  To add to the confusion the guide was written before other tools/plug-in's were made, and the guide has tried to keep up with all the new ways but it's not perfect.  Not to mention, I've tried to accommodate Linux only users, windows only users, etc. So depending on what systems you have access to there are different tools to use.

Converting mp3's to single chapterized m4b is optional, but if you have thousands of books all the mp3 files can really bog down your server. That said, there are many many free apps that can do the conversion, the "[Auto M4B Tool](https://github.com/seanap/auto-m4b)" docker that I made is a Linux utility that automates the conversion.  If using autoM4B then no need to use dropit.

Tags: With the new Audnexus plex plug in the minimum amount of tags you need are correct ALBUM and ARTIST, the plugin will scrape all the other data (cover, series, narrator, etc.).  That said I personally like all of my files to have all tags properly set in the file (in case I use audiobook apps/servers other than plex).  Most CD ripping software lets you set the ALBUM and ARTIST tag, in this case there is no need to do any additional tagging, just move the book to the folder plex watches for Audiobooks.

Flow:
Acquisition > Convert to M4b > Tag > Move to `/audiobooks` folder > use the plex Audnexus plugin

Soup to Nuts (basic):
1. Rip CD 
2. Use mp3tag (or any tagging software) to set the Album/Artist
3. Move tagged book files to your Audiobook folder that plex watches.

Soup to Nuts (Automated Linux/Windows):
1. Rip CD, save book to  `/temp/recently added` folder.  
2. Auto-M4B docker converts to M4B and saves to  `/temp/Untagged` folder. 
3. Using Mp3tag (Windows), use the Audible src script to auto fill in all tag fields and save the tagged file to your plex Audiobook folder.
3. OR using beets.io (Linux) with the Audible plug-in, tag and save the book to your plex Audiobook folder

It sounds like the auto m4b docker isn't working right.  Do you have all the folders created (temp, recently added, merge, untagged, delete, backup, fix)?  It's important to create the temp folder structure exactly like outlined here https://github.com/seanap/auto-m4b

**Reply by yuh8n** [reply]
I did this setup https://github.com/seanap/Auto-M4B-Tool#m4b-tool-automation so I don't have the directory structure outlined in your link.  Even if I create that structure, how do I start using your new M4B tool?  The setup on it is very brief and I'm not sure what to do with the docker that I already setup using the "old" steps...

**Reply by seanap** [reply]
The new docker is way easier.  It's all folder based, the way you use it is to drop your mp3 book into the recently added folder and it does everything in the background and outputs the m4b into the untagged folder.  I would highly recommend using that instead. Easier to setup, use, and manage.

To set up the new autom4b docker you just need to create/edit the docker-compose file with the location of your /temp folder.  If you haven't used docker compose before, you need to install it first, then create a docker-compose.yml file in your home directory copy/paste the code from the instructions (updating the folder paths), then do `docker-compose up -d`

**Reply by yuh8n** [reply]
Ok, so here is what my understanding is:
1 - remove/delete the docker I used using your deprecated instruction set
2 - install the new docker using the below steps from https://github.com/sandreas/m4b-tool#docker
# clone m4b-tool repository
git clone https://github.com/sandreas/m4b-tool.git

# change directory
cd m4b-tool

# build docker image - this will take a while
docker build . -t m4b-tool

# create an alias for m4b-tool running docker
alias m4b-tool='docker run -it --rm -u $(id -u):$(id -g) -v "$(pwd)":/mnt m4b-tool'

# testing the command
m4b-tool --version 

# install all dependencies
sudo apt install ffmpeg mp4v2-utils fdkaac php-cli php-intl php-json php-mbstring php-xml

# install / upgrade m4b-tool
sudo wget https://github.com/sandreas/m4b-tool/releases/download/v.0.4.2/m4b-tool.phar -O /usr/local/bin/m4b-tool && sudo chmod +x /usr/local/bin/m4b-tool

# check installed m4b-tool version
m4b-tool --version
3 - Setup new directory structure
4 - Put new docker-compose.yml with changes (in /home/username/ directory?)

Sorry for the extra questions, trying to put this all together.  Appreciate your patience.

**Comment by seanap** [comment]
Not that docker, that is just the underlying tool.  The way you are setting it up is how I used to use the tool manually (it works but it's way more complicated than it needs to be). All of the steps you outlined are done automatically when using the [Auto-M4B](https://github.com/seanap/auto-m4b) docker 

1. Creat a /temp folder, including all the sub folders outlined in the readme here https://github.com/seanap/auto-m4b/blob/main/README.md#how-to-use
2. Install docker https://docs.docker.com/engine/install/ubuntu/
3. Manage docker as non-root https://docs.docker.com/engine/install/linux-postinstall/
4. Install docker-compose https://docs.docker.com/compose/install/
5. Create the compose file:   
    `nano docker-compose.yml`
6. Paste the yaml code into the compose file, and change the volume mount locations  
7. Put a test mp3 in the /temp/recentlyadded directory.  
8. Start the docker (It should convert the mp3 and leave it in your /temp/untagged directory. It runs automatically every 5 min)  
   `docker-compose up -d`
 
Example yml code (replace the `/mnt/storage/...:` with your actual folder locations, but leave the `:` and everything after):
``` yaml
version: '3.7'
services:
  auto-m4b:
    image: seanap/auto-m4b
    container_name: auto-m4b
    volumes:
      - /mnt/storage/docker/autom4b/config:/config
      - /mnt/storage/temp:/temp  
```

**Reply by yuh8n** [reply]
Ok, I removed the existing docker images and started from scratch with your steps.  I've done up to step 6, creating the yml file in my user home directory, but I don't know where to point my /mnt/storage/docker/autom4b/config path...where in all these steps does your version of autom4b get configured?  Right now I just have a working stock docker after following those outlined steps.

**Reply by seanap** [reply]
I've already configured everything to work right out of the box, you shouldn't need to configure anything.  It will automatically adjust the bitrate to the original source, and it will automatically use all available cpu cores, and it is set up to output all files to a specific directory (/temp/untagged). 

The config directory can be anywhere you like, though I found it works best when located in the local filesystem (as opposed to a network share).  You could even just create a /docker folder in your root directory (eg. /docker/autom4b/config) or you can put it in the srv folder, or even in your home directory. 

Edit: I come from the linuxserver.io world where I run 20+ dockers that do various things.  So just appending any compose code to the end of your docker-compose.yml file will add that service. For example if you also wanted to self host AudiobookShelf server you just add the [compose](https://github.com/advplyr/audiobookshelf/blob/master/docker-compose.yml) (updating your folder locations) and do a `docker-compose up -d` and bobs your uncle.

**Reply by yuh8n** [reply]
Had to run the sudo apt install docker-compose and then everything worked :)  Thank you for your help!  

I'm going to mess around with multiple MP3 files now and see how it does.

**Reply by yuh8n** [reply]
So if I put a bunch of MP3s in there, it is converting each mp3 into an m4b but it's not combining them together into a single file...what did I miss to be able to do that?

**Reply by seanap** [reply]
Do you have all the mp3's in a book folder?  It should be like `/temp/recentlyadded/book/track1.mp3...`

When you rip a CD or download a torrent there should be an option to keep all the files for a book in a parent book folder.  Also, it doesn't handle nested folders very well, so if you can avoid using DISC or CD folders within the book folder, like `/temp/recentlyadded/book/cd1/track1.mp3`  <--- BAD
